### DISCUSSION TITLE
Handles to Native JS Object

### DISCUSSION THREAD

**Discussion by AaronLS** [unoplatform/uno#7605] [Q&A] [answered]
I have been looking at IJSObject types like RenderingLoopAnimator that have a handle to their JS counterpart.  If a C# class like this exposed a method to JS which accepts a native-only JS object, what would that look like?  I.e. if I added a method in C# like this so that I could call from JS and maybe pass a jquery object from JS, what would be the correct type for the `nativeJsHandle` parameter be?

```
public void SomethingCalledFromJS(long nativeJsHandle) 
{
  WebAssemblyRuntime.InvokeJSWithInterop($"{nativeJsHandle}.disable();");

  this.savedHandle = nativeJsHandle;
}
```

**Comment by jeromelaban** [comment, marked as answer]
There's no notion of handle in javacript per say, it's generally a table of IDs and their corresponding javascript objects. If you want to do something similar, you'll have to create a mapping table yourself and use that in the InvokeJS method.

**Reply by AaronLS** [reply]
Is that basically what the `static activeInstances` array is on each TS type?  So what y'all call a "jsHandle" is a long that indexes into an array for each type that holds all of it's own JS object instances?

**Reply by jeromelaban** [reply]
Pretty much, yes. This is the state of interop until GC support comes in WebAssembly, and we can actually get opaque references to "native" objects.

**Reply by AaronLS** [reply]
Thanks so much.
